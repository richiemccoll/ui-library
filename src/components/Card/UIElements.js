import styled from "styled-components";

export const Container = styled.div`
    width: 300px;
    margin: 0 auto;
    box-shadow: 0 12px 15px 0 rgba(0, 0, 0, 0.24), 0 17px 50px 0 rgba(0, 0, 0, 0.19);
    font-family: 'Open Sans', sans-serif;
    margin: .5em;
    padding: .5em;
`;

export const ImageContainer = styled.div`
    width: 100%;
    height: 255px;
    position: relative;
`;

export const Image = styled.img`
    width: 100%;
    height: 100%;
`;

export const Content = styled.div`
    padding: 0 .5em 0 .5em;
`;