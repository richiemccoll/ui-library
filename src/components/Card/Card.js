import React from "react";
import PropTypes from "prop-types";
import { Container, ImageContainer, Image, Content } from './UIElements';

const Card = ({ title, image, description }) => {
  return (
    <Container>
      <ImageContainer>
        <Image src={image} alt={title} />
      </ImageContainer>
      <Content>
        <h2>
          {title}
        </h2>
        <text>
          {description}
        </text>
      </Content>
    </Container>
  );
};

Card.PropTypes = {
  title: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired
};

export default Card;