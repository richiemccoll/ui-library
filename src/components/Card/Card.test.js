import React from "react";
import renderer from "react-test-renderer";
import Card from "./Card";

describe("Card Component", () => {
  it("should render correctly", () => {
    const component = renderer.create(
      <Card
        title="test"
        image="https://i1.sndcdn.com/artworks-000229176048-wbwvfs-t500x500.jpg"
        description="testing the card"
      />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
