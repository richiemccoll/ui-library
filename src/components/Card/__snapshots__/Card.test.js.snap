// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render correctly 1`] = `
<div
  className="sc-bdVaJa iyRwOl"
>
  <div
    className="sc-bwzfXH gOHCgz"
  >
    <img
      alt="test"
      className="sc-htpNat emEaka"
      src="https://i1.sndcdn.com/artworks-000229176048-wbwvfs-t500x500.jpg"
    />
  </div>
  <div
    className="sc-bxivhb gutIee"
  >
    <h2>
      test
    </h2>
    <text>
      testing the card
    </text>
  </div>
</div>
`;
